python/pyproject.toml

```diff
@@ -17,19 +17,21 @@
 
 [build-system]
 requires = [
+    "cmake < 4",
```

Comment:
Why do you need to add a dependency on CMake? Such a dependency is particularly problematic for downstreams, given that regular CMake installations don't provide a `cmake` Python distribution, and following the `build-system.requires` here implies having a non-distribution prebuilt CMake version installed and overriding the correct CMake executable.

If such a dependency is really required, then it should be added via `get_requires_for_build_wheel()` hook if system CMake executable is not provided, much like `scikit-build-core` does.

Reply:
Ah nice catch - I think this is a relic from the original design which also let you build Arrow C++ from source and use it as a subproject. We scrapped that in favor of requiring Arrow C++ to be installed for PyArrow to build, so this can likely be removed